DEVELOPMENT_SETUP.md

```diff
@@ -0,0 +1,43 @@
+\# Local Development Setup Issues and Solutions
+
+
+
+\## Problems Encountered During Initial Setup
+
+
+
+1\. \*\*Dependency Conflicts\*\*: The `req.txt` files contain conflicting package versions, particularly between `spacy` and `thinc`
+
+2\. \*\*Missing SSH Keys\*\*: PDF parser component tries to clone private repositories via SSH
+
+3\. \*\*Complex Project Structure\*\*: Two separate projects with different setup requirements
+
+4\. \*\*Unclear Installation Instructions\*\*: No clear minimal setup path for development
+
+
+
+\## Solutions
+
+
+
+\### Minimal Working Setup
```

Comment:
this part might be useful.  only check in things that like the new dev would use and follow.